src/cmakeTaskProvider.ts

```diff
@@ -219,37 +339,86 @@ class CustomBuildTaskTerminal implements vscode.Pseudoterminal, proc.OutputConsu
             }
             this.closeEmitter.fire(0);
         } catch {
+            this.writeEmitter.fire(localize("build.finished.with.error", "Build finished with error(s).") + endOfLine);
             this.closeEmitter.fire(-1);
         }
     }
 
-    private async runConfigTask(): Promise<any> {
-        this.writeEmitter.fire(localize("config.started", "Config Started...") + endOfLine);
-        const result: number | undefined =  await vscode.commands.executeCommand('cmake.configure');
-        this.closeEmitter.fire(result ? result : -1);
-    }
-
     private async runInstallTask(): Promise<any> {
-        this.writeEmitter.fire(localize("install.started", "Install Started...") + endOfLine);
-        const result: number | undefined =  await vscode.commands.executeCommand('cmake.install');
-        this.closeEmitter.fire(result ? result : -1);
+        this.writeEmitter.fire(localize("install.started", "Install task started...") + endOfLine);
+        this.targetIsIgnored();
+        const cmakeTools: CMakeTools | undefined = this.getCMakeTools();
+        if (!cmakeTools) {
+            return;
+        }
+        const result: number | undefined =  await cmakeTools.runBuild(['install'], false, this);
+        if (result === undefined || result === null) {
```

Comment:
`null` is not one of the type declarations for `result`. Do you need to check it?